CONTRIBUTING.md

```diff
@@ -9,70 +36,163 @@ license the code under the same terms as the project's overall 'outbound'
 license - in our case, this is almost always Apache Software License v2 (see
 [LICENSE](LICENSE)).
 
-## How to contribute
+# 2. What do I need?
+
+The code of Synapse is written in Python 3. To do pretty much anything, you'll need [a recent version of Python 3](https://wiki.python.org/moin/BeginnersGuide/Download).
+
+The source code of Synapse is hosted on Github. You will also need [a recent version of git](https://github.com/git-guides/install-git).
+
+For some tests, you will need [a recent version of Docker](https://docs.docker.com/get-docker/).
+
+## Under Unix (macOS, Linux, BSD, ...)
+
+Once you have installed Python 3, please open a Terminal and run:
+
+```sh
+mkdir -p ~/synapse
+python3 -m venv ./env
+source ./env/bin/activate
+pip install -e ".[all,lint,mypy,test,black,tox]"
+```
+
+This will install the developer dependencies for the project.
+
+## Under Windows
+
+TBD
+
+
+# 3. Get the source.
 
 The preferred and easiest way to contribute changes is to fork the relevant
 project on github, and then [create a pull request](
 https://help.github.com/articles/using-pull-requests/) to ask us to pull your
 changes into our repo.
 
-Some other points to follow:
+Please base your changes on the `develop` branch.
+
+```sh
+git clone git@github.com:matrix-org/synapse.git
+```
```

Comment:
I think this actually gives the wrong advice since it won't pull from their fork?